### PR TITLE
deploy CSIDriver object api version dynamically based on kubeVersion

### LINKED ISSUE #887: CSIDriver v1 is not available in 1.17
/kind bug

The chart indicates >=1.17 compatibility. And yet in 1.17 in storage v1 CSIDriver is not available.
Either discontinue support for 1.17 and bump the chart to 1.18 minimum or bust down  resource version to v1beta1

The incoming fix will do the latter.

### PR DESCRIPTION
**Pull request by AndyXiangLi**
Is this a bug fix or adding new feature?
Fixes #887
What is this PR about? / Why do we need it?
storage.k8s.io/v1beta1 CSIDriver is deprecated in v1.19+, unavailable in v1.22+;
CSIDriver is general available in storage.k8s.io/v1 since v1.19
Instead of using v1beta1 version for all helm deployment, check the kubeVersion first in helm deployment and use v1 for clusterVersion >=1.19 and v1beta1 for clusterVersion >=1.17 and < 1.19

What testing is done?
Tested in 1.17, 1.18, 1.19 cluster

### CODE REVIEW

File: charts/aws-ebs-csi-driver/templates/csidriver.yaml
Comment (krmichel) [inline charts/aws-ebs-csi-driver/templates/csidriver.yaml:1]: I thought this became [GA in 1.18](https://kubernetes-csi.github.io/docs/csi-driver-object.html)

**Review by krmichel — CHANGES_REQUESTED** [review]

File: charts/aws-ebs-csi-driver/templates/csidriver.yaml
Comment (AndyXiangLi) [reply charts/aws-ebs-csi-driver/templates/csidriver.yaml:1]: Yeah, v1beta1 is deprecated in 1.19 so I used 1.19 here.. 
I can change it to 1.18 to sync with your PR, or do you want to cherry-pick the helm chart update to the release branch after merge? Then I will close this on as well

Comment (krmichel) [reply charts/aws-ebs-csi-driver/templates/csidriver.yaml:1]: Go ahead and change it to 1.18 just to get it fixed in the release branch faster.  Thanks

Comment (AndyXiangLi) [reply charts/aws-ebs-csi-driver/templates/csidriver.yaml]: Updated, can I get a lgtm?

**Review by krmichel — APPROVED** [review]

**Pull request merged**
